### PR TITLE
Rotate off python3.5 support

### LINKED ISSUE #112: Drop py35
In https://github.com/UNFmontreal/Dcm2Bids/pull/111/checks?check_run_id=1808751063

> py35 installed: DEPRECATION: Python 3.5 reached the end of its life on September 13th, 2020. Please upgrade your Python as Python 3.5 is no longer maintained. pip 21.0 will drop support for Python 3.5 in January 2021. pip 21.0 will remove support for this functionality.,astroid==2.4.2,attrs==20.3.0,bids-validator==1.5.10,click==7.1.2,coverage==5.4,dcm2bids @ file:///home/runner/work/Dcm2Bids/Dcm2Bids/.tox/.tmp/package/1/dcm2bids-2.1.6.dev2%2Bgfe41a18.zip,docopt==0.6.2,flake8==3.8.4,future==0.18.2,importlib-metadata==2.1.1,iniconfig==1.1.1,isort==4.3.21,lazy-object-proxy==1.4.3,mccabe==0.6.1,nibabel==3.0.2,num2words==0.5.10,numpy==1.18.5,packaging==20.9,pandas==0.25.3,pathlib2==2.3.5,patsy==0.5.1,pluggy==0.13.1,py==1.10.0,pybids==0.12.4,pycodestyle==2.6.0,pyflakes==2.2.0,pylint==2.6.0,pyparsing==2.4.7,pytest==6.1.2,pytest-black==0.3.12,pytest-cov==2.11.1,pytest-flake8==1.0.7,pytest-pylint==0.18.0,python-dateutil==2.8.1,pytz==2021.1,scipy==1.4.1,six==1.15.0,SQLAlchemy==1.3.22,toml==0.10.2,typed-ast==1.4.2,wrapt==1.12.1,zipp==1.2.0

Py35 failed the build. Weird. Have no PRs gone in since September? I was sure I did something in like..November? Maybe tox only recently hard-deprecated it?

~~Blocks #110.~~

### PR DESCRIPTION
**Pull request by kousu**
Fixes #112.

### CODE REVIEW

**Comment by kousu** [comment]
It's surprising to me that there's both:

https://github.com/UNFmontreal/Dcm2Bids/blob/38f803f4b7f74eea2c350e2a16589b6c804d2afe/.github/workflows/tests.yml#L16

and

https://github.com/UNFmontreal/Dcm2Bids/blob/38f803f4b7f74eea2c350e2a16589b6c804d2afe/tox.ini#L10-L14

oh and 

https://github.com/UNFmontreal/Dcm2Bids/blob/38f803f4b7f74eea2c350e2a16589b6c804d2afe/tox.ini#L7

Some of these must be redundant. Why use tox at all if gh-actions lets you cover different envs? It seems that in practice it's not a big deal because each tox run ends up only trying one version of python. That's weird, isn't it? Maybe I don't understand tox. In fact I'm sure I don't understand tox, and maybe this is just the conventional way to use it, but I it's sort of weird isn't it?

**Comment by arnaudbore** [comment]
Hi Nick I was also sceptical but find this [documentation](https://github.com/ymyzk/tox-gh-actions)
I also tried a couple of options and it seems to be the way tox and gh actions are working together. Can you add python 3.9 also ?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
           - ubuntu-latest  # ubuntu-18.04
           - macos-latest  # macOS-10.14
           - windows-latest  # windows-2019
-        python-version: [3.5, 3.6, 3.7]
+        python-version: [3.6, 3.7, 3.8]
```
Comment (arnaudbore) [inline .github/workflows/tests.yml]:
```suggestion
        python-version: [3.6, 3.7, 3.8, 3.9]
```

File: setup.py [hunk 2]
```diff
@@ -55,9 +55,9 @@ def install_requires():
     "Operating System :: Microsoft :: Windows",
     "Operating System :: Unix",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3.5",
     "Programming Language :: Python :: 3.6",
     "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
```
Comment (arnaudbore) [inline setup.py:60]:
```suggestion
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
```

File: tox.ini [hunk 3]
```diff
@@ -4,22 +4,22 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py35, py36, py37
-; envlist = clean, py35, py36, py37, report
+envlist = py36, py37, py38
+; envlist = clean, py36, py37, py38, report
```
Comment (arnaudbore) [inline tox.ini]:
```suggestion
; envlist = clean, py36, py37, py38, py39, report
```

File: tox.ini [hunk 4]
```diff
@@ -4,22 +4,22 @@
...
 
 [gh-actions]
 python =
-    3.5: py35
     3.6: py36
     3.7: py37
+    3.8: py38
```
Comment (arnaudbore) [inline tox.ini:14]:
```suggestion
    3.8: py38
    3.9: py39
```

File: tox.ini [hunk 5]
```diff
@@ -4,22 +4,22 @@
...
 
 [testenv]
 deps = -rrequirements-test.txt
 commands = pytest --cov --cov-report=xml -s
 ; commands = pytest --cov --cov-append --black --flake8 --pylint
 ; depends =
-;     {py35, py36, py37}: clean
-;     report: py35, py36, py37
+;     {py36, py37, py38}: clean
+;     report: py36, py37, py38
```
Comment (arnaudbore) [inline tox.ini]:
```suggestion
;     report: py36, py37, py38, py39
```

**Review by arnaudbore — COMMENTED** [review]
Add python 3.9

**Comment by kousu** [comment]
> Hi Nick I was also sceptical but find this [documentation](https://github.com/ymyzk/tox-gh-actions)
> I also tried a couple of options and it seems to be the way tox and gh actions are working together.

Huh. Peu bizarre but alright, thanks for double checking that.

> Can you add python 3.9 also ?

C'est une vraie bonne idée!

File: tox.ini [hunk 6]
```diff
@@ -4,22 +4,22 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py35, py36, py37
-; envlist = clean, py35, py36, py37, report
+envlist = py36, py37, py38
```
Comment (kousu) [inline tox.ini]:
```suggestion
envlist = py36, py37, py38, py39
```

**Comment by cbedetti** [comment]
Hi,
The tests were quick to run and I was running them locally before the implementation of gh-actions.
Also, this implementation was done in the early days of gh-actions. Maybe that's why.

**Pull request merged**